posts/2022-06-01-user-feature-install.adoc

```diff
@@ -0,0 +1,381 @@
+
+== Creating a user feature
+
+In this blog, we will use Eclipse to create a Maven project. For more information, check out link:{url-prefix}/guides/maven-intro.html[Building a web application with Maven].
+
+Alternatively, you can link:https://start.openliberty.io[create and download starter application] from the Open Liberty website.
+
+
+In this post, we walk through the following steps of developing a user feature:
+
+- Create an OSGi bundle containing your Java classes using link:https://felix.apache.org/documentation/subprojects/apache-felix-maven-bundle-plugin-bnd.html#_using_the_plugin[Apache Felix Maven Bundle Plugin (BND)]
+- Package your OSGi bundle to Enterprise Subsystem Archive (ESA) file with link:https://aries.apache.org/documentation/modules/esamavenpluginproject.html[ESA Maven Plugin]
+- Create a Bill of Materials (BOM) for your ESA file
+- Install your Open Liberty user feature onto Open Liberty runtime by using link:https://github.com/OpenLiberty/ci.maven[Liberty Maven Plugin]
+
+'''
+=== Developing an OSGi bundle with simple activation
+
+The most straightforward way to control the lifecycle of your OSGi bundle code is to implement the `org.osgi.framework.BundleActivator` interface in one of the classes within your bundle. The `BundleActivator` start and stop methods are called when the server starts or stops.
+
+[start=1]
+. Create a Maven project with your desired Maven coordinates.
++
+[source, txt]
+----
+Group Id : test.user.test.osgi
+Artifact Id : SimpleActivator
+Version : 0.0.1-SNAPSHOT
+----
+
+. Add the `org.osgi.core` dependency
++
+[source, xml]
+----
+<dependency>
+    <groupId>org.osgi</groupId>
+    <artifactId>org.osgi.core</artifactId>
+    <version>6.0.0</version>
+</dependency>
+----
+
+. Create an implementation for the OSGi `BundleActivator` class in the project
++
+[source, java]
+----
+package com.example.bundle;
+
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+
+public class Activator implements BundleActivator {
+	public void start(BundleContext context) throws Exception {
+		System.out.println("Sample bundle starting");
+		// Insert bundle activation logic here
+	}
+
+	public void stop(BundleContext context) throws Exception {
+		System.out.println("Sample bundle stopping");
+		// Insert bundle deactivation logic here
+	}
+}
+----
+
+. Build the Bundle
++
+To use the maven-bundle-plugin, explicitly specify `packaging` as `bundle` in the `pom.xml` file.
++
+[source, xml]
+----
+<packaging>bundle</packaging>
+----
++
+We also need to specify the configuration. The `Bundle-SymbolicName` is the only required header for most cases but may require other OSGi headers depending on your implementation.
```

Comment:
How do they know whether their implementation requires other OSGi headers or not? And which headers they might need?